fedimint-client/src/transaction/sm.rs

```diff
@@ -54,7 +56,7 @@ pub enum TxSubmissionStates {
     ///
     /// **This state is final**
     Rejected(TransactionId, String),
-    /// this should never happen with a honest federation and bug-free code
+    /// this state is currently never assumed as all errors are retried
```

Comment:
Maybe [mark it as deprecated](https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-deprecated-attribute) so we get warning if it's being used?

Reply:
done